pkg/apis/serving/v1/revision_lifecycle.go

```diff
@@ -200,6 +200,13 @@ func (rs *RevisionStatus) PropagateAutoscalerStatus(ps *autoscalingv1alpha1.PodA
 		// that implies that |service.endpoints| > 0.
 		rs.MarkResourcesAvailableTrue()
 		rs.MarkContainerHealthyTrue()
+
+		// Mark resource unavailable if we are scaling back to zero, but we never achieved the required scale
+		// and deployment status was not updated properly by K8s. For example due to an image pull error.
+		if ps.ScaleTargetNotScaled() {
+			condScaled := ps.GetCondition(autoscalingv1alpha1.PodAutoscalerConditionScaleTargetScaled)
+			rs.MarkResourcesAvailableFalse(condScaled.Reason, condScaled.Message)
```

Comment:
I think I'd prefer to keep ResourcesAvailable condition separate from this one.

I associate ResourcesAvailable whether resources can be created and programmed